__tests__/happy.postActivity.js

```diff
@@ -0,0 +1,63 @@
+import 'dotenv/config';
+
+import onErrorResumeNext from 'on-error-resume-next';
+
+import { DirectLine } from '../src/directLine';
+import { timeouts } from './constants.json';
+import * as createDirectLineOptions from './setup/createDirectLineOptions';
+import postActivity from './setup/postActivity';
+import waitForBotEcho from './setup/waitForBotEcho';
+import waitForConnected from './setup/waitForConnected';
+
+describe('Happy path', () => {
+  let unsubscribes;
+
+  beforeEach(() => unsubscribes = []);
+  afterEach(() => unsubscribes.forEach(fn => onErrorResumeNext(fn)));
+
+  describe('should connect, send message, and receive echo from bot', () => {
+    let directLine;
+
+    describe('using REST', () => {
+      beforeEach(() => jest.setTimeout(timeouts.rest));
+
+      test('with secret', async () => {
+        directLine = new DirectLine(await createDirectLineOptions.forREST({ token: false }));
+      });
+
+      test('with token', async () => {
+        directLine = new DirectLine(await createDirectLineOptions.forREST({ token: true }));
+      });
+    });
+
+    // test('using Streaming Extensions', async () => {
```

Comment:
I'm now assuming these are left on purpose. Please verify then feel free to ignore xD